action/protocol/staking/candidate_statemanager.go

```diff
@@ -51,7 +51,7 @@ type (
 )
 
 // NewCandidateStateManager returns a new CandidateStateManager instance
-func NewCandidateStateManager(sm protocol.StateManager) (CandidateStateManager, error) {
+func NewCandidateStateManager(sm protocol.StateManager, enableSMStorage bool) (CandidateStateManager, error) {
```

Comment:
nit: as you so accurately said, csm's lifespan is linked to the ctx.
maybe pass in ctx as a general parameter, if sth is needed in future, there's no need to modify the input list again